native/arrow_format_nif/src/utils.rs

```diff
@@ -1,3 +1,292 @@
 use std::collections::HashMap;
 
+use crate::message::record_batch::{Buffer, Compression, FieldNode, RecordBatch};
+use crate::message::schema::field::{Dictionary, Field, Name};
+use crate::message::schema::types::{FixedSizeList, Int, Type};
+use crate::message::schema::{Endianness, Feature, Schema};
+use crate::message::{Body, Header, Message, Version};
+
+#[cfg(test)]
+use arrow_format::ipc;
+
 pub type CustomMetadata = Vec<HashMap<String, String>>;
+
+/// Returns an example `Message` of a `Schema`
+pub fn schema() -> Message {
```

Comment:
I generally don't like "utils" as a module name because it tends to become a blackhole of all things that don't quite have a place.

If I understood correctly, `schema` and `message_batch` a just dummy functions that will be replaced where they are currently called by actual code in the future.

If this is indeed the case, I think it would be best if they were prefixed with `dummy_` and set in the `message` module instead.

Reply:
They are functions which return testing data, and are used solely in testing.
Presently they are are used in the `test_decode` function in `crate`, and in various Rust specific tests.

Should I still prefix them with `dummy` and move them or is keeping them in `utils` or in a `test_utils` good enough?

Reply:
Should `test_decode` actually be exposed in the NIF?

Reply:
Yes, it's a function to test if everything decodes properly into Erlang.

This PR only implements the encoding part of this NIF, and I will have to deal with decoding later.

Reply:
So that function will be removed as soon as the decoding is implemented, right?
I believe it's fine to leave it where it is then :)

Reply:
I've put the `arrow_*` into a `test::fixtures` module in `lib.rs`. I hoping this is what you had in mind when you said `test::fixtures`? I wasn't able to find a way to keep them in a file in the `tests` dir and use them in unit tests in the `src` dir.